Modules/syslogmodule.c

```diff
@@ -380,6 +397,29 @@ syslog_exec(PyObject *module)
     return 0;
 }
 
+static int
+_syslog_traverse(PyObject *module, visitproc visit, void *arg)
+{
+    _syslog_state *state = get_syslog_state(module);
+    Py_VISIT(state->S_ident_o);
+    return 0;
+}
+
+static int
+_syslog_clear(PyObject *module)
+{
+    _syslog_state *state = get_syslog_state(module);
+    Py_CLEAR(state->S_ident_o);
```

Comment:
Implementations of openlog(3) might just store the "ident" argument (which is why S_ident_o exists). On such implementations the Py_CLEAR call results into use-after-free when syslog() is called from C code after this point.

Probably best to call closelog(3) with some default arguments before clearing.